lib/internal/modules/cjs/loader.js

```diff
@@ -806,7 +807,7 @@ Module.prototype._compile = function(content, filename) {
     result = compiledWrapper.call(thisValue, exports, require, module,
                                   filename, dirname);
   }
-  if (depth === 0) stat.cache = null;
+  if (requireDepth === 0) statCache = null;
   return result;
 };
 
```

Comment:
I noticed with the removed tests that this change makes ensuring require depth is tracked no longer possible? Is there a way to have both?

Reply:
The former tests did not notice the regression and as such did not test the actual behavior. The only way to actually test this is to expose some information that is also available to users and I would like to prevent that. That's why I removed the tests without adding a new one.